### ISSUE TITLE
Typescript version incompatible

### ISSUE THREAD

**Issue by dapplion** [bbc/typescript-docs-verifier#1]
Running on a project with a newer Typescript version causes issues since this package declares Typescript as a direct dependency.

```
/home/lion/Code/eth2.0/lodestar/node_modules/typescript-docs-verifier/node_modules/ts-node/src/index.ts:261
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
../../tsconfig.build.json(15,5): error TS5023: Unknown compiler option 'useUnknownInCatchVariables'.
    at createTSError (/home/lion/Code/eth2.0/lodestar/node_modules/typescript-docs-verifier/node_modules/ts-node/src/index.ts:261:12)
    at Object.register (/home/lion/Code/eth2.0/lodestar/node_modules/typescript-docs-verifier/node_modules/ts-node/src/index.ts:265:42)
    at new TypeScriptRunner (/home/lion/Code/eth2.0/lodestar/node_modules/typescript-docs-verifier/src/TypeScriptRunner.ts:7:12)
    at new SnippetCompiler (/home/lion/Code/eth2.0/lodestar/node_modules/typescript-docs-verifier/src/SnippetCompiler.ts:29:19)
    at Object.compileSnippets (/home/lion/Code/eth2.0/lodestar/node_modules/typescript-docs-verifier/index.ts:19:20)
    at Object.<anonymous> (/home/lion/Code/eth2.0/lodestar/node_modules/typescript-docs-verifier/bin/compile-typescript-docs.ts:58:24)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47
```

*from https://github.com/ChainSafe/lodestar/pull/3097/checks?check_run_id=3555738804*

`useUnknownInCatchVariables` was added in Typescript 4.4.0

Thanks for the tool tho!

**Comment by paulbrimicombe** [comment]
Thanks for getting in touch @dapplion — I'll try to take a look at this soon.

**Comment by dapplion** [comment]
> Thanks for getting in touch @dapplion — I'll try to take a look at this soon.

Listing the typescript dependencies as peer dependencies would solve the problem. But then I'm not sure if the tool would work on its own. Maybe ship two versions, one that's stand-alone, and another with peer dependencies?

Bumping Typescript version would solve the problem right now, but won't be a sustainable solution long term

**Comment by paulbrimicombe** [comment]
@dapplion sorry for the delay. I've just published a release candidate with a fix for this (see https://github.com/bbc/typescript-docs-verifier/pull/2). Could you install `typescript-docs-verifier@next` and check that it works for you please? It has `typescript` as a `peerDependency` so you'll have to make sure that is installed as well (which it probably is for a TypeScript project).

**Comment by dapplion** [comment]
Thank you so much, will try the new version asap

**Comment by paulbrimicombe** [comment]
Did you get the chance to test this @dapplion ? I'm probably going to publish this as version 2.0.0 in the next few days.

**Comment by dapplion** [comment]
> Did you get the chance to test this @dapplion ? I'm probably going to publish this as version 2.0.0 in the next few days.

Thanks for the ping just pushed a PR https://github.com/ChainSafe/lodestar/pull/3313 waiting for CI

**Comment by paulbrimicombe** [comment]
I've released version 2.0.0 now which should fix this issue. Thanks for getting in touch @dapplion.
